src/test/java/io/airlift/slice/TestSlices.java

```diff
@@ -67,6 +80,52 @@ private static void testWrapping(ByteBuffer buffer)
         }
     }
 
+    @Test
+    public void testWrapAllTypes()
+    {
+        boolean[] booleanArray = {true, false, false, true, false, true};
+        assertEquals(wrappedBooleanArray(booleanArray).getByte(0) == 1, booleanArray[0]);
+        assertEquals(wrappedBooleanArray(booleanArray, 2, 4).getByte(0) == 1, booleanArray[1]);
```

Comment:
Can you make all these `1, 4` so the slice is in the middle of the array?

Also, with the current code, this should be `booleanArray[2]`, but it will be correct after updating the slice.